inst/sql/sql_server/subsets/LimitSubsetOperator.sql

```diff
@@ -19,7 +19,7 @@ FROM (
       AND DATEDIFF(day, t.cohort_start_date, op.observation_period_end_date) >= @follow_up_time
   }
 ) c
-{@limit_to == 'firstEver' | @limit_to == 'lastEver'}?{
+{@limit_to == 'firstEver' | @limit_to == 'lastEver' | @limit_to == 'all'}?{
```

Comment:
I know you made this change here, but I'm looking at the existing code for this question:

I think i understand that if you specify a limit of `firstEver` or `lastEver` (and now all) you're going to restrict it to only consider records in the same observation period as the event. via the join to OP....I'm wondering why this is?

My understanding of firstEver vs. first remiainign is where you check the ordinal = 1.   If it is first ever, then you rownumber the raw cohort table, pick first per person, and your'e done.  If you pick earliestRemaining, then you apply the prior observation criteria (via the where) first to get rid of things without necessary prior observation, and then you apply the where ordinal=1 on that remaining set.  So the crux of my qustion is it seems you need observatino period when you have a prior observation requirement, but first vs. remaining is simply a mater of when you look for ordinal = 1

Reply:
The use of the observation period table is  to allow the `prior_time` and `follow_up_time` parameters in conjunction with `first_ever/last_ever` and now "all". 

I think the requirement of entries to overlap with the observation time was to stop this calculation being negative but I'm not sure now why it was implemented this way.  Perhaps it was to censor events outside of observation perioids but I'm not sure, it looks like the condition could be removed.

Reply:
Yeah, that's my thought too. It makes sense to bring in observation period when you need to restrict by prior observation period > 0.   Otherwise, it doesn't matter.   So, I don't want to throw a huge wrench in the work here, if it works, it works.  But, it is not logically consistent to bring in the observation period into the query unless you need to do something like `datediff(d, op_start_date, cohort_start_date) > @priorObservationDays`

Reply:
good point - I have updated the query to skip the joins if the prior/follow up are set to 0